pkg/plugins/repo/ifaces.go

```diff
@@ -6,34 +6,90 @@ import (
 	"strings"
 )
 
-// Service is responsible for retrieving plugin information from a repository.
+// Service is responsible for retrieving plugin archive information from a repository.
 type Service interface {
 	// GetPluginArchive fetches the requested plugin archive.
 	GetPluginArchive(ctx context.Context, pluginID, version string, opts CompatOpts) (*PluginArchive, error)
 	// GetPluginArchiveByURL fetches the requested plugin from the specified URL.
 	GetPluginArchiveByURL(ctx context.Context, archiveURL string, opts CompatOpts) (*PluginArchive, error)
-	// GetPluginDownloadOptions fetches information for downloading the requested plugin.
-	GetPluginDownloadOptions(ctx context.Context, pluginID, version string, opts CompatOpts) (*PluginDownloadOptions, error)
+	// GetPluginArchiveInfo fetches information needed for downloading the requested plugin.
+	GetPluginArchiveInfo(ctx context.Context, pluginID, version string, opts CompatOpts) (*PluginArchiveInfo, error)
 }
 
 type CompatOpts struct {
-	GrafanaVersion string
-	OS             string
-	Arch           string
+	grafanaVersion string
+	system         SystemCompatOpts
+}
+
+type SystemCompatOpts struct {
+	os   string
+	arch string
+}
+
+func (co CompatOpts) GrafanaVersion() (string, bool) {
+	if len(co.grafanaVersion) > 0 {
+		return co.grafanaVersion, true
+	}
+	return "", false
+}
+
+func (co CompatOpts) System() (SystemCompatOpts, bool) {
+	os, osSet := co.system.OS()
+	arch, archSet := co.system.Arch()
+	if !osSet && !archSet {
```

Comment:
This should probably be `||` as I think it will always make sense to have these specific together

Reply:
sounds reasonable to me